docs/jax-101/02-jitting.ipynb

```diff
@@ -0,0 +1,570 @@
+      "100 loops, best of 5: 7.9 ms per loop\n"
+     ]
+    }
+   ],
+   "source": [
+    "import jax\n",
+    "import jax.numpy as jnp\n",
+    "\n",
+    "def selu(x, alpha=1.67, lambda_=1.05):\n",
+    "  return lambda_ * jnp.where(x > 0, x, alpha * jnp.exp(x) - alpha)\n",
+    "\n",
+    "x = jnp.arange(1000000)\n",
+    "%timeit selu(x).block_until_ready()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "ecN5lEXe6ncy"
+   },
+   "source": [
+    "The code above is sending one operation at a time to the accelerator. This limits the ability of the XLA compiler to optimize our functions.\n",
+    "\n",
+    "Naturally, what we want to do is give the XLA compiler as much code as possible, so it can fully optimize it. The jax transformation **jit** does that. The example below shows how to use jit to speed up the previous function."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "id": "nJVEwPcH6bQX",
+    "outputId": "64b5eeab-0861-4cac-9c48-34586aa14fb6"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "1000 loops, best of 5: 1.6 ms per loop\n"
+     ]
+    }
+   ],
+   "source": [
+    "selu_jit = jax.jit(selu)\n",
+    "\n",
+    "# Warm up\n",
+    "selu_jit(x).block_until_ready()\n",
+    "\n",
+    "%timeit selu_jit(x).block_until_ready()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "hMNKi1mYXQg5"
+   },
+   "source": [
+    "Here's what just happened:\n",
+    "\n",
+    "1) We defined `selu_jit` as the compiled version of `selu`.\n",
+    "\n",
+    "2) We ran `selu_jit` once on `x`. This is where JAX does its tracing -- it needs to have some inputs to wrap in tracers, after all. The jaxpr is then compiled using XLA into very efficient code optimized for your GPU or TPU. Subsequent calls to `selu_jit` will now use that code, skipping our old python implementation entirely. \n",
+    "\n",
+    "(If we didn't include the warm-up call separately, everything would still work, but then the compilation time would be included in the benchmark. It would still be faster, because we run many loops in the benchmark, but it wouldn't be a fair comparison.)\n",
+    "\n",
+    "3) We timed the execution speed of the compiled version. (Note the use of `block_until_ready()`, which is required due to JAX's [Asynchronous execution](https://jax.readthedocs.io/en/latest/async_dispatch.html) model)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "DRJ6R6-d9Q_U"
+   },
+   "source": [
+    "## Why can't we just jit everything?\n",
+    "\n",
+    "After seeing the example above, you might be wondering whether we should simply apply `jax.jit` to every function. To understand why this is not the case, and when we should/shouldn't apply jit, let's first check some cases where jit doesn't work."
```

Comment:
Suggestion: (again, "apply jit" -> "apply `jit`" as in a method/fn `jit`. Using  https://jax.readthedocs.io/en/latest/notebooks/thinking_in_jax.html#to-jit-or-not-to-jit as a reference)

```suggestion
    "After going through the example above, you might be wondering whether we should simply apply `jax.jit` to every function. To understand why this is not the case, and when we should/shouldn't apply `jit`, let's first check some cases where `jit` doesn't work."
```